intelmq/bots/collectors/microsoft_dcu/collector.py

```diff
@@ -0,0 +1,44 @@
+from intelmq.lib.bot import Bot, sys
+from intelmq.lib.message import Event
+from intelmq.bots import utils
+
+from azure.storage import BlobService
+
+import gzip
+import StringIO
+from urlparse import urlparse
+
+
+class DCUCollectorBot(Bot):
+    """ 
+      This IntelMQ collector is for getting a blob
+      from an azure account (Microsoft dcu).
+      It opens the account and reads all containers.
+    """
+
+    def process(self):
+        account_name = self.parameters.azure_account_name
+        account_key = self.parameters.azure_account_key
+
+        blob_service = BlobService(account_name, account_key, protocol="https")
+        proxy_options = urlparse(self.parameters.https_proxy or "")
+
+        if proxy_options.port and proxy_options.hostname:
+            blob_service.set_proxy(host=proxy_options.hostname, port=proxy_options.hostname)
```

Comment:
here we have a problem: port = proxy_options.hostname ... that can't work.
Also please note that the proxy_options object contains "None:None" at this stage - always.